### PR TITLE
Fix a bug when using where in and enum arrays

### LINKED ISSUE #112: Where-In queries with enum-arrays will generate bad requests
Given the following test cases:

```
[Fact]
public void ExpressionEnumWhereInList()
{
    List<OrderState> orderStates = new[] { OrderState.Open, OrderState.Complete, OrderState.Confirmed }.ToList();
    Expression<Func<Order, bool>> expression = x => x.OrderState.In(orderStates);
    IQueryPredicateExpressionVisitor queryPredicateExpressionVisitor = this.linqFixture.GetService<IQueryPredicateExpressionVisitor>();
    string result = queryPredicateExpressionVisitor.Render(expression);
    Assert.Equal("orderState in (\"Open\", \"Complete\", \"Confirmed\")", result);
}

[Fact]
public void ExpressionEnumWhereInListValueOf()
{
    List<OrderState> orderStates = new[] { OrderState.Open.valueOf(), OrderState.Complete.valueOf(), OrderState.Confirmed.valueOf() }.ToList();
    Expression<Func<Order, bool>> expression = x => x.OrderState.In(orderStates);
    IQueryPredicateExpressionVisitor queryPredicateExpressionVisitor = this.linqFixture.GetService<IQueryPredicateExpressionVisitor>();
    string result = queryPredicateExpressionVisitor.Render(expression);
    Assert.Equal("orderState in (\"Open\", \"Complete\", \"Confirmed\")", result);
}

[Fact]
public void ExpressionEnumWhereInArray()
{
    OrderState[] orderStates = new[] { OrderState.Open, OrderState.Complete, OrderState.Confirmed };
    Expression<Func<Order, bool>> expression = x => x.OrderState.In(orderStates);
    IQueryPredicateExpressionVisitor queryPredicateExpressionVisitor = this.linqFixture.GetService<IQueryPredicateExpressionVisitor>();
    string result = queryPredicateExpressionVisitor.Render(expression);
    Assert.Equal("orderState in (\"Open\", \"Complete\", \"Confirmed\")", result);
}

[Fact]
public void ExpressionEnumWhereInArrayValueOf()
{
    OrderState[] orderStates = new[] { OrderState.Open.valueOf(), OrderState.Complete.valueOf(), OrderState.Confirmed.valueOf() };
    Expression<Func<Order, bool>> expression = x => x.OrderState.In(orderStates);
    IQueryPredicateExpressionVisitor queryPredicateExpressionVisitor = this.linqFixture.GetService<IQueryPredicateExpressionVisitor>();
    string result = queryPredicateExpressionVisitor.Render(expression);
    Assert.Equal("orderState in (\"Open\", \"Complete\", \"Confirmed\")", result);
}
```

The tests will all fail with something like this:

image

I already provided a fix, I don't know if this is the right way, but the tests won't fail, and our internal application tests also like this fix.

### PR DESCRIPTION
**Pull request by btastic**
Fixes #112

### CODE REVIEW

File: commercetools.Sdk/commercetools.Sdk.Linq/Query/Converters/CollectionPredicateVisitorConverter.cs [hunk 1]
```diff
@@ -47,6 +48,22 @@ public IPredicateVisitor Convert(Expression expression, IPredicateVisitorFactory
...
 
+        private static bool IsStringOrEnum(object item, out string result)
+        {
+            switch (item)
+            {
+                case string stringValue:
+                    result = stringValue.WrapInQuotes();
+                    return true;
+                case Enum enumValue:
+                    result = enumValue.ToString().WrapInQuotes();
```
Comment (MicheleRezk) [inline commercetools.Sdk/commercetools.Sdk.Linq/Query/Converters/CollectionPredicateVisitorConverter.cs]: it's better to use description enum attribute if exists to be like:
result = enumValue.GetDescription().WrapInQuotes();

Comment (btastic) [reply commercetools.Sdk/commercetools.Sdk.Linq/Query/Converters/CollectionPredicateVisitorConverter.cs]: Fixed

File: commercetools.Sdk/Tests/commercetools.Sdk.Linq.Tests/QueryPredicateTests.cs [hunk 2]
```diff
@@ -750,5 +749,45 @@ public void ExpressionWithBirthDateFilters()
...
+            IQueryPredicateExpressionVisitor queryPredicateExpressionVisitor = this.linqFixture.GetService<IQueryPredicateExpressionVisitor>();
+            string result = queryPredicateExpressionVisitor.Render(expression);
+            Assert.Equal("orderState in (\"Open\", \"Complete\", \"Confirmed\")", result);
+        }
+
+        [Fact]
+        public void ExpressionEnumWhereInListValueOf()
+        {
+            List<OrderState> orderStates = new[] { OrderState.Open.valueOf(), OrderState.Complete.valueOf(), OrderState.Confirmed.valueOf() }.ToList();
+            Expression<Func<Order, bool>> expression = x => x.OrderState.In(orderStates);
```
Comment (MicheleRezk) [inline commercetools.Sdk/Tests/commercetools.Sdk.Linq.Tests/QueryPredicateTests.cs]: I think we're using valueOf extension with inline variables, so i think it should be like this:
Expression<Func<Order, bool>> expression = x => x.OrderState.In(new[] { OrderState.Open, OrderState.Complete, OrderState.Confirmed }.ToList().valueOf());

File: commercetools.Sdk/Tests/commercetools.Sdk.Linq.Tests/QueryPredicateTests.cs [hunk 3]
```diff
@@ -750,5 +749,45 @@ public void ExpressionWithBirthDateFilters()
...
+            IQueryPredicateExpressionVisitor queryPredicateExpressionVisitor = this.linqFixture.GetService<IQueryPredicateExpressionVisitor>();
+            string result = queryPredicateExpressionVisitor.Render(expression);
+            Assert.Equal("orderState in (\"Open\", \"Complete\", \"Confirmed\")", result);
+        }
+
+        [Fact]
+        public void ExpressionEnumWhereInArrayValueOf()
+        {
+            OrderState[] orderStates = new[] { OrderState.Open.valueOf(), OrderState.Complete.valueOf(), OrderState.Confirmed.valueOf() };
+            Expression<Func<Order, bool>> expression = x => x.OrderState.In(orderStates);
```
Comment (MicheleRezk) [inline commercetools.Sdk/Tests/commercetools.Sdk.Linq.Tests/QueryPredicateTests.cs]: I think the same is here for valueOf

File: commercetools.Sdk/Tests/commercetools.Sdk.Linq.Tests/QueryPredicateTests.cs (hunk 2 above)
Comment (btastic) [reply commercetools.Sdk/Tests/commercetools.Sdk.Linq.Tests/QueryPredicateTests.cs]: Sure thing! Updated..

File: commercetools.Sdk/Tests/commercetools.Sdk.Linq.Tests/QueryPredicateTests.cs (hunk 3 above)
Comment (btastic) [reply commercetools.Sdk/Tests/commercetools.Sdk.Linq.Tests/QueryPredicateTests.cs]: Updated it, too

**Pull request merged**